crates/pallet-transaction-fees/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+
+    /// Temporary value (cleared at block finalization) which contains current block author, so we
+    /// can issue rewards during block finalization.
+    #[pallet::storage]
+    pub(super) type BlockAuthor<T: Config> = StorageValue<_, T::AccountId>;
+
+    /// Temporary value (cleared at block finalization) which contains current block storage fees,
+    /// so we can issue rewards during block finalization.
+    #[pallet::storage]
+    pub(super) type CollectedStorageFees<T: Config> = StorageValue<_, BalanceOf<T>>;
+
+    /// Temporary value (cleared at block finalization) which contains current block compute fees,
+    /// so we can issue rewards during block finalization.
+    #[pallet::storage]
+    pub(super) type CollectedComputeFees<T: Config> = StorageValue<_, BalanceOf<T>>;
+
+    /// Temporary value (cleared at block finalization) which contains current block tips, so we can
+    /// issue rewards during block finalization.
+    #[pallet::storage]
+    pub(super) type CollectedTips<T: Config> = StorageValue<_, BalanceOf<T>>;
+
+    /// Pallet rewards for issuing rewards to block producers.
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    /// `pallet-transaction-fees` events
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Storage fees escrow change.
+        StorageFeesEscrowChange {
+            /// State of storage fees escrow before block execution.
+            before: BalanceOf<T>,
+            /// State of storage fees escrow after block execution.
+            after: BalanceOf<T>,
+        },
+        /// Storage fees.
+        StorageFeesReward {
+            /// Receiver of the storage fees.
+            who: T::AccountId,
+            /// Amount of collected storage fees.
+            amount: BalanceOf<T>,
+        },
+        /// Compute fees.
+        ComputeFeesReward {
+            /// Receiver of the compute fees.
+            who: T::AccountId,
+            /// Amount of collected compute fees.
+            amount: BalanceOf<T>,
+        },
+        /// Tips.
+        TipsReward {
+            /// Receiver of the tip.
+            who: T::AccountId,
+            /// Amount of collected tips.
+            amount: BalanceOf<T>,
+        },
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T>
+    where
+        BalanceOf<T>: From<u8> + From<u64>,
+    {
+        fn on_initialize(now: BlockNumberFor<T>) -> Weight {
+            Self::do_initialize(now);
+            T::WeightInfo::on_initialize()
+        }
+
+        fn on_finalize(now: BlockNumberFor<T>) {
+            Self::do_finalize(now);
+        }
+    }
+}
+
+impl<T: Config> Pallet<T>
+where
+    BalanceOf<T>: From<u64>,
+{
+    fn do_initialize(_n: T::BlockNumber) {
+        let block_author: T::AccountId = frame_system::Pallet::<T>::digest()
```

Comment:
I suggest introducing a trait `FindAuthor`(don't have to reuse the one from substrate) and using an associated type  
 whenever a pallet needs the block author, even currently this piece of code is only duplicated twice.